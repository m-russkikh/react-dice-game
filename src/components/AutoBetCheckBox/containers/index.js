export { default as AutoBetCheckBoxContainer } from './AutoBetCheckBoxContainer';
