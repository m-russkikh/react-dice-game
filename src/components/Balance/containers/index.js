export { default as BalanceContainer } from './BalanceContainer';
