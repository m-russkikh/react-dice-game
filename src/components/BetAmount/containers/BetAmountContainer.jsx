import React, { Component } from 'react';

import InputControl from '../../InputControl';
import { BetAmountContainerPropType } from '../propTypes';


class BetAmountContainer extends Component {
  handleChange = ({ currentTarget: { value } }) => {
    this.props.setBetAmount(parseInt(value, 10));
  }

  render() {
    return (
      <InputControl
        label="Bet Amount"
        onChange={this.handleChange}
      />
    );
  }
}

BetAmountContainer.propTypes = BetAmountContainerPropType;

export default BetAmountContainer;
