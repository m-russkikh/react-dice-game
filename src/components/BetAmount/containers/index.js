export { default as BetAmountContainer } from './BetAmountContainer';
