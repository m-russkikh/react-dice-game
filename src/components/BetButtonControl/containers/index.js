export { default as BetButtonControlContainer } from './BetButtonControlContainer';
