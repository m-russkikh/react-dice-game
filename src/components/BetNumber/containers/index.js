export { default as BetNumberContainer } from './BetNumberContainer';
