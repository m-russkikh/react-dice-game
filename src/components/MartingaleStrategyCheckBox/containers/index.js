export { default as MartingaleStrategyCheckBoxContainer } from './MartingaleStrategyCheckBoxContainer';
