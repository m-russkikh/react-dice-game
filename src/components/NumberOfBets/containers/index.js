export { default as NumberOfBetsContainer } from './NumberOfBetsContainer';
