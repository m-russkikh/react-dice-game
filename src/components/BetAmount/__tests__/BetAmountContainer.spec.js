import { getShallowWrapper } from '../../../utils/testUtils';
import { BetAmountContainer } from '../containers';


describe('BetAmountContainer', () => {
  let props;
  let wrapper;
  let instance;

  beforeEach(() => {
    props = {
      setBetAmount: jest.fn(),
    };

    wrapper = getShallowWrapper(BetAmountContainer, props);
    instance = wrapper.instance()
  });

  describe('render', () => {
    it('should display correct', () => {
      expect(wrapper.debug()).toMatchSnapshot();
    });
  });

  describe('methods', () => {
    describe('handleChange', () => {
      it('should call setBetAmount', () => {
        const event = { currentTarget: { value: 5 } };

        instance.handleChange(event);

        expect(props.setBetAmount).toHaveBeenCalledTimes(1);
        expect(props.setBetAmount).toHaveBeenCalledWith(event.currentTarget.value);
      });
    });
  });
});
