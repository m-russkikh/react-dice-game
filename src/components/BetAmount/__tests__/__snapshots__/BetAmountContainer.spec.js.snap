// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BetAmountContainer render should display correct 1`] = `"<InputControlView label=\\"Bet Amount\\" onChange={[Function]} />"`;
